src/transformers/models/llama/modeling_llama.py

```diff
@@ -378,16 +380,12 @@ def forward(
 
         kv_seq_len = key_states.shape[-2]
         if past_key_value is not None:
-            kv_seq_len += past_key_value[0].shape[-2]
+            kv_seq_len += past_key_value.get_seq_length(self.layer_idx)
```

Comment:
```suggestion
             if self.layer_idx is None:
                 raise ValueError(f"The cache structure has changed since version XXX. If you are using {self.__class__.__name__} for auto-regressive decoding with k/v caching, please make sure to initialize the attention class with a layer index.")
            kv_seq_len += past_key_value.get_seq_length(self.layer_idx)
```